tests/bsim/bluetooth/mesh/src/test_lcd.c

```diff
@@ -255,32 +290,40 @@ static void test_cli_split_comp_data_request(void)
 		.data = &srv_rsp_comp_2,
 	};
 
-	bt_mesh_device_setup(&prov, &comp_1);
+	bt_mesh_device_setup(&prov, (comp_page == 0 || comp_page == 128) ? &comp_1 : &comp_2);
+	bt_mesh_comp2_register(&comp_p2);
 	prov_and_conf(cli_cfg);
-	target_node_alloc(comp_1, srv_cfg);
-
-	offset = 0;
+	target_node_alloc((comp_page == 0 || comp_page == 128) ? comp_1 : comp_2, srv_cfg);
 
 	/* Get local data */
-	err = bt_mesh_comp_data_get_page_0(&local_comp, offset);
+	err = bt_mesh_comp_data_get_page(&local_comp, comp_page, 0);
 	/* Operation is successful even if all data cannot fit in the buffer (-E2BIG) */
 	if (err && err != -E2BIG) {
-		FAIL("CLIENT: Failed to get comp data Page 0: %d", err);
+		FAIL("CLIENT: Failed to get comp data Page %d: %d", err, comp_page);
 	}
-	total_size = bt_mesh_comp_page_0_size();
+
+	uint16_t total_size = bt_mesh_comp_page_size(comp_page);
+
+	/* Verify that the total comp page size is not larger than the provided buffer */
+	ASSERT_OK(total_size <= CONFIG_BT_MESH_COMP_PST_BUF_SIZE);
+
+	/* Wait a bit until the server is ready to respond*/
+	k_sleep(K_SECONDS(2));
 
 	/* Get first server composition data sample and verify data */
-	ASSERT_OK(bt_mesh_large_comp_data_get(0, SRV_ADDR, page, offset, &srv_rsp_1));
-	rsp_equals_local_data_assert(SRV_ADDR, &srv_rsp_1, &local_comp, page, offset, total_size,
-				     &prev_len);
+	ASSERT_OK(bt_mesh_large_comp_data_get(0, SRV_ADDR, comp_page, offset, &srv_rsp_1));
+
```

Comment:
Please remove the added empty line here and below. Lets move style changes to a separate commit/PR so it is easier to review.